### PR TITLE
extend pingmachine to support telegraf measurement export

### PR DESCRIPTION
**Pull request by haymanni**

### CODE REVIEW

File: lib/Pingmachine/Order.pm [hunk 1]
```diff
@@ -272,6 +288,53 @@ sub _update_last_results_file {
     };
 }
 
+# Sends messages to telegraf
+sub _update_telegraf {
+    my ($self, $rrd_time, $results) = @_;
+
+    my @rtts    = @{$results->{pings}};
+    my @sorted_rtts    = @{$results->{rtts}};
```
Comment (nimdanitro) [inline lib/Pingmachine/Order.pm:296]: please add an explicit sort here. 
`my @sorted_rtts    = sort @{$results->{rtts}};`

File: lib/Pingmachine/Order.pm [hunk 2]
```diff
@@ -272,6 +288,53 @@ sub _update_last_results_file {
...
+            PeerPort => $telegraf_port,
+            Proto    => 'udp',
+            Type     => IO::Socket::SOCK_DGRAM,
+            Blocking => 0,
+        )  or die("Can't open UDP socket: $@");
+
+        # set our socket to non blocking mode
+        AnyEvent::Util::fh_nonblocking($telegraf_socket, 1);
+
+        my $result_rrd_time = sprintf("%d%09d", $rrd_time , ($rrd_time - int($rrd_time)) * 1_000_000_000);
```
Comment (nimdanitro) [inline lib/Pingmachine/Order.pm]: can you add a comment here why this NS time conversion is required.

**Review by nimdanitro — CHANGES_REQUESTED** [review]
can you also adjust the README.md file and document the telegraf file content?

**Comment by schweikert** [comment]
Nice work! I am happy to see pingmachine alive.

**Review by nimdanitro — APPROVED** [review]
Nice work @haymanni

Thanks a lot for your contribution.

**Pull request merged**